test/unit/browser-test.js

```diff
@@ -0,0 +1,54 @@
+define([
+    'utils/browser'
+], function (browser) {
+    /* jshint qunit: true */
+
+    module('browser');
+
+    var testerGenerator = function (assert, method) {
+        return function (left, right, message) {
+            assert.strictEqual(method.apply(this, left), right, message);
+        };
+    };
+
+    test('browser.isInt', function(assert) {
+        var test = testerGenerator(assert, browser.isInt);
+        test([0],       true);
+        test([0x10],    true);
+        test([24],      true);
+        test([0.5],     false);
+        test(['10.'],    true);
+        test(['3.1'],   false);
+        test([NaN],     false);
+        test([null],    false);
+        test([undefined], false);
+    });
+
+    test('browser checks', function(assert) {
+        var fakeVersion = 'fakeVersion12';
```

Comment:
This could be a phantomjs only test, using fake user agents -- not sure if it's possible though.